### ISSUE TITLE
[Question] How could I get event which triggered the method #onError(error, stackTrace)?

### ISSUE THREAD

**Issue by dtvtkien** [felangel/bloc#3006]
I have several use cases in my project which could throw my custom exception. Each use case is presented as an Event in a specific Bloc. I need to find which event throw exceptions.
So far I'm using `#onEvent(event)` to store the current event and `#onError(error, stackTrace)` to catch the exception.
But when I migrate to Bloc v8.0.1 (which I find useful to make some events concurrently and repeatable) then the stored event in `#onEvent(event)` might not be the last event that thrown the exception.
How could I handle this problem? For example, could you add an `event` parameter to [here](https://github.com/felangel/bloc/blob/35599cbe37973645e6d4c7466da33c0c9556ddd6/packages/bloc/lib/src/bloc.dart#L477) - such as `onError(event, error, stackTrace)`?

**Comment by felangel** [comment]
Hi @dtvtkien 👋 
Thanks for opening an issue!

Are you able to provide a minimal reproduction sample to illustrate the specific use case you have? I’m happy to take a closer look. Thanks!

**Comment by dtvtkien** [comment]
Thanks for your fast response @felangel!

Unfortunately, I'm in my vacation so I'm not able to create a sample. 

My use case is: After I got the mushroom species from the identifier engine, I'd like to `concurrently` call Wikipedia API (to fetch summary information) and Take Care API (to fetch watering/harvesting/pruning, etc of this mushroom). I used 2 separate events to handle them.

The issue is 1 of them might throw an HTTP exception but I don't want to extract `host` or `endpoint` to detect which event caused the error.

Before Bloc v8.0, my `BaseBloc` abstract class  is:
```dart
@override
  Stream<BaseState> mapEventToState(BaseEvent event) async* {
    _currentEvent = event;
    if (event is FetchWikipedia) {
      yield* handleFetchWikipedia(event as FetchWikipedia);
      return;
    }
    if (event is FetchCareInfo) {
      yield* handleFetchCareInfo(event as FetchCareInfo);
      return;
    }
    yield* handleEvent(event).handleError(
      (exception) => handleError(
        exception,
        tag: event.tag,
      ),
    );
  }
```
And because events are triggered sequentially so I could easily handle this issue (I know which event caused the exception by `event.tag`).

And now I've just migrated to Bloc v8.0 with concurrent event handler buckets, so how I could detect the event that throws exceptions?
